<launch>

    <!-- An example which reads a CM-0120 CO2 sensor -->
    <!--
    <node name="co2_sensor_reader" pkg="nodelet" type="nodelet" 
          args="standalone simple_serial_port_reader/SimpleSerialPortReader" output="screen">
        <remap from="formatted" to="co2/state"/>
        <param name="device" value="/dev/ttyUSB0"/>
        <param name="baud_rate" value="9600"/>
        <param name="match_expression" value="Z (\d+) z (\d+)\r\n"/>
        <param name="format_expression" value="CO2: Filtered = $1 ppm, Raw = $2 ppm"/>
        <param name="verbose" value="false"/>
    </node>
    -->

    <!-- An example which reads a CM-0202 O2 sensor -->
    <node name="o2_sensor_reader" pkg="nodelet" type="nodelet" 
          args="standalone simple_serial_port_reader/SimpleSerialPortReader" output="screen">
        <remap from="formatted" to="o2/state"/>
        <param name="device" value="/dev/ttyUSB0"/>
        <param name="baud_rate" value="9600"/>
        <param name="match_expression" value="% (\d+.\d+) e \d+\r\n"/>
        <param name="format_expression" value="O2 = $1 %"/>
        <param name="verbose" value="true"/>
    </node>

</launch>